src/components/externalLinks/ExternalLinksCRUD/IdentifierSelector.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useEffect, useState } from 'react'
+import { components, InputActionMeta } from 'react-select'
+import CreatableSelect from 'react-select/creatable'
+import { Checkbox, CHECKBOX_VALUE, ClearIndicator, MultiValueRemove, noop } from '../../common'
+import { ExternalLinkIdentifierType, ExternalLinkScopeType, IdentifierSelectorProps } from '../ExternalLinks.type'
+import { ReactComponent as AddIcon } from '../../../assets/icons/ic-add.svg'
+import { IdentifierSelectStyles } from '../ExternalLinks.utils'
+
+export default function IdentifierSelector({
+    index,
+    link,
+    selectedIdentifiers,
+    clusters,
+    allApps,
+    handleLinksDataActions,
+    getErrorLabel,
+}: IdentifierSelectorProps) {
+    const [identifierSearchInput, setIdentifierSearchInput] = useState('')
+
+    useEffect(() => {
+        clearIdentifierSearchInput()
+    }, [link.type])
+
+    const identifierMenuList = (props): JSX.Element => {
+        return (
+            <components.MenuList {...props}>
+                {link.type === ExternalLinkScopeType.AppLevel && (
+                    <>
+                        {identifierSearchInput ? (
+                            <div className="flex left pl-8 pt-6 pb-6" onClick={markOptionAsExternalApp}>
+                                <AddIcon className="icon-dim-16 fcb-5 mr-8" />
+                                <span className="fs-13 fw-4 lh-20 cb-5">
+                                    External helm app ‘{identifierSearchInput}’
+                                </span>
+                            </div>
+                        ) : (
+                            <div className="cn-5 pl-8 pt-6 pb-6 dc__italic-font-style">
+                                Enter app name for externally deployed helm apps
+                            </div>
+                        )}
+                    </>
+                )}
+                {props.children}
+            </components.MenuList>
+        )
+    }
+
+    const noMatchingIdentifierOptions = (): string => {
```

Comment:
We should declare this in helper i guess. it has been used on multiple places

Reply:
Sure @vivek-devtron. I'll try to cover this in one of the upcoming PRs 🙂